rabbitmq/resource_binding.go

```diff
@@ -64,6 +64,21 @@ func resourceBinding() *schema.Resource {
 	}
 }
 
+// Because slashes are used to separate different components when constructing binding IDs,
+// we need a way to ensure any components that include slashes can survive the round trip.
+// Percent-encoding is a straightforward way of doing so.
+// (reference: https://developer.mozilla.org/en-US/docs/Glossary/percent-encoding)
+
+func percentEncodeSlashes(s string) string {
+	// Encode any percent signs, then encode any forward slashes.
+	return strings.Replace(strings.Replace(s, "%", "%25", -1), "/", "%2F", -1)
+}
+
+func percentDecodeSlashes(s string) string {
+	// Decode any forward slashes, then decode any percent signs.
+	return strings.Replace(strings.Replace(s, "%2F", "/", -1), "%25", "%", -1)
+}
```

Comment:
I think we should move these two functions to the `util.go` file. Any objections?

Reply:
I wasn't _confident_ these would be generally useful since they only encode & decode a select couple of characters, but I do not have a strong opinion. I'll happily defer to you.